source/E2E.Tests/Services/BesiegerCamps/BesiegerCampFieldTests.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Common.Util;
+using E2E.Tests.Environment;
+using E2E.Tests.Environment.Instance;
+using HarmonyLib;
+using TaleWorlds.CampaignSystem.Party;
+using TaleWorlds.CampaignSystem.Siege;
+using Xunit.Abstractions;
+
+namespace E2E.Tests.Services.BesiegerCamps;
+
+public class BesiegerCampFieldTests : IDisposable
+{
+    private E2ETestEnvironment TestEnvironment { get; }
+
+    private EnvironmentInstance Server => TestEnvironment.Server;
+
+    private IEnumerable<EnvironmentInstance> Clients => TestEnvironment.Clients;
+
+    private readonly string BesiegerCampId;
+
+    public BesiegerCampFieldTests(ITestOutputHelper output)
+    {
+        TestEnvironment = new E2ETestEnvironment(output);
+
+        var BesiegerCamp = ObjectHelper.SkipConstructor<BesiegerCamp>();
+
+        // Create BesiegerCamp on the server
+        Assert.True(Server.ObjectManager.AddNewObject(BesiegerCamp, out BesiegerCampId));
+
+        // Create BesiegerCamp on all clients
+        foreach (var client in Clients)
+        {
+            var clientBesiegerCamp = ObjectHelper.SkipConstructor<BesiegerCamp>();
+            Assert.True(client.ObjectManager.AddExisting(BesiegerCampId, clientBesiegerCamp));
+        }
+    }
+
+    public void Dispose()
+    {
+        TestEnvironment.Dispose();
+    }
+
+    [Fact]
+    public void ServerChangeBesiegerCampLeaderParty()
+    {
+        // Arrange
+        var field = AccessTools.Field(typeof(BesiegerCamp), nameof(BesiegerCamp._leaderParty));
+        var intercept = TestEnvironment.GetIntercept(field);
+
+        // Create garrison instances on server
+        Assert.True(Server.ObjectManager.AddNewObject(ObjectHelper.SkipConstructor<MobileParty>(), out var mobilePartyId));
+
+        // Create garrison instances on all clients
+        foreach (var client in Clients)
+        {
+            var clientParty = ObjectHelper.SkipConstructor<MobileParty>();
+            Assert.True(client.ObjectManager.AddExisting(mobilePartyId, clientParty));
+        }
```

Comment:
This does the exact same unless you need empty fields

```suggestion
        // Create garrison instances on server
        TestEnvironment.CreateRegisteredObject<MobileParty>();
```